### ISSUE TITLE
JPA Reference implementation

### ISSUE THREAD

**Issue by maybeec** [devonfw-forge/devonfw-microservices#24]
devonfw uses spring data + queryDSL with reflection

* https://github.com/devonfw/devon4j/blob/master/documentation/guide-jpa.asciidoc
* https://github.com/devonfw/devon4j/blob/master/documentation/guide-jpa-query.asciidoc
* https://github.com/devonfw/devon4j/blob/master/documentation/guide-jpa-performance.asciidoc
* https://github.com/devonfw/devon4j/blob/master/documentation/guide-repository.asciidoc
* https://github.com/devonfw/devon4j/blob/master/documentation/guide-dao.asciidoc

tkit utilizes DAOs

How to combine? What will be the proposed solution for the future on GraalVM? 
Will the throw away queryDSL or switch to code generation?
Is there support of spring data for quarkus? Will we choose it again over DAOs?

**Comment by GuentherJulian** [comment]
I think this one can be closed @maybeec. We adapted the devon4j [repository](https://github.com/devonfw/devon4j/blob/master/documentation/guide-repository.asciidoc) and [query](https://github.com/devonfw/devon4j/blob/master/documentation/guide-jpa-query.asciidoc#dynamic-queries) guides, so that the matches with Spring and Quarkus. We also created a separate guide showing the QueryDsl integration of devon4j for Spring (https://github.com/devonfw/devon4j/blob/master/documentation/spring/guide-querydsl-spring.asciidoc).
So to summarize we recommend Spring Data and QueryDsl also for Quarkus. Is there anything further to do for this issue?

**Comment by maybeec** [comment]
Where can I find now how to utilize queryDSL with Quarkus? I cannot see it unfortunately!?

**Comment by GuentherJulian** [comment]
We have at least changed the part about dynamic queries in the [query guide](https://github.com/devonfw/devon4j/blob/master/documentation/guide-jpa-query.asciidoc#dynamic-queries), so that this part is valid for both Spring **and** Quarkus using Querydsl with generated metaclasses (no devon4j module, this is described in the separate guide). But you are right, I think we should also mention how to integrate querydsl and querydsl annotation processor and maybe how to create expressions in more detail.

**Comment by GuentherJulian** [comment]
Created https://github.com/devonfw/devon4j/issues/482 to extend the guide
